segmentstore/server/src/main/java/io/pravega/segmentstore/server/security/Channels.java

```diff
@@ -0,0 +1,131 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.server.security;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import io.netty.channel.Channel;
+import io.netty.channel.group.ChannelGroup;
+import io.netty.channel.group.ChannelGroupFuture;
+import io.netty.channel.group.DefaultChannelGroup;
+import io.netty.util.concurrent.GlobalEventExecutor;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+/**
+ * Encapsulates a {@link ChannelGroup} object for maintaining a group of {@link Channel} objects and common operations
+ * performed on the group.
+ */
+@Slf4j
+public final class Channels {
+
+    private final Lock sequential = new ReentrantLock();
+
+    // See info about ChannelGroups here: https://netty.io/4.1/api/io/netty/channel/group/ChannelGroup.html
+    private ChannelGroup channels =
+            new DefaultChannelGroup(GlobalEventExecutor.INSTANCE, true);
+
+    /**
+     * Adds the specified channel to the group.
+     *
+     * @param channel {@code channel} to be added
+     *
+     */
+    public void add(Channel channel) {
+        Preconditions.checkNotNull(channel);
+        sequential.lock();
+        try {
+            channels.add(channel);
+            log.debug("Added channel [{}] to the group.", channel);
+        } finally {
+            sequential.unlock();
+        }
+    }
+
+    /**
+     * Clears the current channels held by this object.
+     *
+     */
+    public void clear() {
+        if (!this.channels.isEmpty()) {
```

Comment:
This should be under the lock too?

Reply:
I removed this class now, as it is irrelevant given the change in strategy in favor of `SslContext` reload.